### PR TITLE
Updates to handle resuming INACTIVE actions/flows

### PR DESCRIPTION
**Pull request by jimPruyne**
The main new feature is addition of the resume method on both actions
and flows (as usual, flows version built in terms of the action
version). This will check for the special INACTIVE condition based on
a `code` property in the action's details for a `required_scope`
value. If present, it will walk the user through the process of
creating new consents for the `required_scope` value prior to invoking
the resume method on the flow/action.

Other big changes:

* We've imported a new method of managing tokens / directing the user
for consent. This moves away from the FRL package and uses a direct
implementation on the Globus SDK native client. It also, therefore,
prompts on the command line for cut-and-paste to/from a browser rather
than trying to launch a browser directly. Lessons learned from work on
the Timer CLI.

* We've added methods for managing a session including `whoami`,
`logout` and `revoke`. This is particularly helpful in development
when managing tokens and being able to re-set to a known "ground
state" when issuing tokens, etc. See `globus-automate session --help`
for more info.

### CODE REVIEW

**Comment by urimandujano** [comment]
Got this after doing a pull, checkout, and `poetry install`. It's probably just a missed import statement but I'm not sure if there's more than that going on

```
❯ globus-automate session --help
Traceback (most recent call last):
  File "/Users/uriel/Projects/github.com/globus/globus-automate-client/.venv/bin/globus-automate", line 2, in <module>
    from globus_automate_client.cli.main import app
  File "/Users/uriel/Projects/github.com/globus/globus-automate-client/globus_automate_client/cli/main.py", line 7, in <module>
    from globus_automate_client.cli import actions, flows, queues, session
  File "/Users/uriel/Projects/github.com/globus/globus-automate-client/globus_automate_client/cli/flows.py", line 35, in <module>
    from globus_automate_client.token_management import CLIENT_ID
  File "/Users/uriel/Projects/github.com/globus/globus-automate-client/globus_automate_client/token_management.py", line 7, in <module>
    from fair_research_login import ConfigParserTokenStorage, NativeClient
ModuleNotFoundError: No module named 'fair_research_login'
```

File: globus_automate_client/token_management.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import os
 from typing import Optional
 
-from fair_research_login import ConfigParserTokenStorage, NativeClient
-from fair_research_login.exc import AuthFailure, LocalServerError
 from globus_sdk import AccessTokenAuthorizer, GlobusAPIError
 from globus_sdk.exc import AuthAPIError
 
+from fair_research_login import ConfigParserTokenStorage, NativeClient
+from fair_research_login.exc import AuthFailure, LocalServerError
```
Comment (urimandujano) [inline globus_automate_client/token_management.py:8]: This is probably what's preventing me from running `globus-automate session`, weird that our editors just decided to move the import around. There don't seem to be any other diffs in this file so let's make sure we drop all references to the library. I'm not even sure if we need this file with the new module but I defer on that to you.

File: pyproject.toml [hunk 2]
```diff
@@ -36,7 +36,7 @@ python = "^3.6"
 globus-sdk = "^2.0"
 graphviz = "^0.12"
 typer = {extras = ["all"], version = "^0.3.0"}
-fair-research-login = {version = "^0.2.0"}
+# fair-research-login = {version = "^0.2.0"}
```
Comment (urimandujano) [inline pyproject.toml:39]: I'm ok with dropping this completely.

File: globus_automate_client/flows_client.py [hunk 3]
```diff
@@ -454,6 +454,31 @@ def flow_action_status(
...
+        Resume a Flow Action which is in an INACTIVE state.
+
+        :param flow_id: The UUID identifying the Flow which triggered the Action
+
+        :param flow_scope: The scope associated with the Flow ``flow_id``. If
+            not provided, the SDK will attempt to perform an introspection on
+            the Flow to determine its scope automatically.
+
+        :param flow_action_id: The ID specifying the Action for which's status
+            we want to query
```
Comment (urimandujano) [inline globus_automate_client/flows_client.py:470]: Maybe just an update to the phrasing here - "the Action to resume execution"

File: globus_automate_client/cli/auth.py [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+    native_client = _get_globus_sdk_native_client(client_id, client_name)
+
+    for scope in scopes:
+        token_set = token_cache.get_tokens(scope)
+        if token_set is not None:
+            token_sets[scope] = token_set
+        else:
+            needed_scopes.add(scope)
+
+    if len(needed_scopes) > 0 and not no_login:
```
Comment (urimandujano) [inline globus_automate_client/cli/auth.py:203]: What happens if we need scopes, but no_login = False. I guess the return would never contain the authorizer for the scope, I'm just wondering if that's how it should be handled.

File: globus_automate_client/cli/actions.py [hunk 5]
```diff
@@ -176,6 +176,58 @@ def action_status(
...
+            result = ac.status(action_id)
+            body = result.data
+            status = body.get("status")
+            details = body.get("details", {})
+            code = details.get("code")
+            if status == "INACTIVE" and code == "ConsentRequired":
+                required_scope = details.get("required_scope")
+                if required_scope is not None:
+                    ac = create_action_client(action_url, action_scope=required_scope)
+        result = ac.resume(action_id)
```
[the remark this answers is not in the document]
Comment (jimPruyne) [reply globus_automate_client/cli/actions.py:225]: That's basically the plan. I think you followed the flow here completely.

File: pyproject.toml (hunk 2 above)
Comment (jimPruyne) [reply pyproject.toml:39]: I have done that now.

File: globus_automate_client/cli/auth.py (hunk 4 above)
Comment (jimPruyne) [reply globus_automate_client/cli/auth.py:202]: yeah, I think the intention here is that in this case, we'll try to get something from the tokens we loaded from cache. In all likelihood, that will not find anything, and we just end up returning the empty `authorizers` dict.

**Comment by jimPruyne** [comment]
@urimandujano Updated from review. Also realized I forgot to add a file. Oops...

Anyway, ready for re-review and potentially release?

**Comment by urimandujano** [comment]
> Also realized I forgot to add a file.

Happens to the best of us (:

For some reason this still isn't working for me, possibly the result of a weird merge?

```
❯ globus-automate --help
Traceback (most recent call last):
  File "/Users/uriel/Projects/github.com/globus/globus-automate-client/.venv/bin/globus-automate", line 2, in <module>
    from globus_automate_client.cli.main import app
  File "/Users/uriel/Projects/github.com/globus/globus-automate-client/globus_automate_client/cli/main.py", line 7, in <module>
    from globus_automate_client.cli import actions, flows, queues, session
  File "/Users/uriel/Projects/github.com/globus/globus-automate-client/globus_automate_client/cli/actions.py", line 209, in <module>
    ActionOutputFormat.json,
NameError: name 'ActionOutputFormat' is not defined
```

**Review by urimandujano — APPROVED** [review]

**Pull request merged**